practices/appeng/decision-guide.md

```diff
@@ -0,0 +1,200 @@
+and Truss organization
+
+## Intended Audience
+
+While we're here to focus on the "why",
+taking a step back and focusing on the "who"
+is a prequisite to making decision.
+Within the context of an ADR,
+it's important to enumerate who a decision benefits
+and draw back to that group within decision factors.
+Be curious about what that group's needs are
+and how the values in this guide intertwine with them.
+
+Commonly,
+decisions will affect the following groups:
+
+1. The users of a service.
+   Depending on the makeup of a project team,
+   this will include other client stakeholders.
+1. The team building a service.
+   If the team building the service includes client stakeholders
+   (ex. a product owner involved in development),
+   this group may encompass them.
+1. When creating practice level ADRs,
+   we're adding in one more group to focus on:
+   the AppEng practice.
+   This is mentioned in the
+   "What makes this decision generally applicable" section
+   of the [template](./adrs/template.md).
+
+Here is an example drawing back to those groups
+in an ADR focused on picking a new database:
+
+> Currently, the org's database isn't easily learnable,
+> causing difficult onboarding for developers
+> and slow releases for database related features
+> and bugfixes
+> to users.
+
+And for the practice:
+
+> Most (if not all) of our projects use a database,
+> and commonly they are relational.
+> To make onboarding between projects easier
+> and to generate better shared learning resources,
+> deciding on a common relational database for the practice
+> will increase effeciency delivering on all projects.
+
+## Engineering Values and Decision Factors
+
+Now that we've established a clear context and user group,
+we can apply a set of common values
+to aid in making a decision.
+
+Note that these values won't always apply.
+They're outlined as a guide for our work,
+not an exhaustive list.
+A single decision likely won't tick all these boxes.
+In addition,
+client or individual use case may have different needs.
+Rely on these as a default
+and be sure to review the context for your decision
+for making exceptions.
+
+Keeping with the above example
+(choosing a relational store),
+we'll follow an example for each value,
+prefacing with this statement:
+
+> We chose to standardize on PostgreSQL
+> as a default relational database for the practice.
+> ...
+
+### User Friendly
+
+Drawing from our team charter,
+"Software is only as valuable as the people it helps".
+We seek to make decisions and use tools
+that drive end user value
+and usability.
+Regardless of how cutting edge,
+organized,
+or fast
+our software is underneath the hood,
+if a user can't access it
+and perform its intended purpose with ease,
+we have not built good software.
+This includes making our software accessible
+to our users,
+stakeholders,
+and team.
+
+> ... Postgres internal types align well with common API level types
```

Comment:
Open to a better example here.

Reply:
I really like the Postgres examples in the doc, although agree that this one feels a little forced and is more specific to public API consumers.  I'm also not sure what types Postgres has that give it a significant advantage in this respect over other major relational databases.  That said, I'm not sure I have a great example either, but I'll toss out a few thoughts:

- Performance-related examples?  Not sure if Postgres has significant advantages in that realm, but seems like something that could impact an end user.
- Lower (or non-existent) licensing/support costs?  Other commercial databases like Oracle can cost a fortune for a license as well as ongoing support.  End users may not directly care about this, but it can lower the cost of building/maintaining software.
- More security transparency being open source?  That may be a stretch too, but end users might indirectly benefit from that as well.

Reply:
Better defaults/fewer traps, maybe? Postgres enforces stricter SQL conformance than MySQL does by default, which we believe means fewer errors in production; as I recall it also defaults to better settings for character sets, collation, and time/date fields.

Postgres does let you define your own column level types in ways that MySQL doesn't, although I'm not sure if that's necessarily a compelling argument here (although, PostGIS is *amazing*).